src/utils/hooks/use-redirect-location.js

```diff
@@ -1,11 +1,22 @@
 import { useCallback, useEffect, useState } from 'react';
 import { useLocation, matchPath } from 'react-router-dom';
-import { routes, params } from '../../config';
+import { routes, params, tabLabels } from '../../config';
+import { useGeneratePathnameForExperimentTracking } from './use-generate-pathname';
+
+const getDefaultTabLabel = (searchParams) => {
+  // If the view from URL is not matched the tabLabels
+  // then set the default value to be the first one from tabLabels
+  return tabLabels.includes(searchParams.view)
+    ? searchParams.view
+    : tabLabels[0];
+};
```

Comment:
Try and keep comments above the function name. I would also suggest we start adding in docstring comments as this part of the application grows in complexity. It'll be easier to manage.

```suggestion
/**
 * If the view from URL is not matched the tabLabels then set the default
 * value to be the first one from tabLabels
 * @param {object} searchParams 
 * @returns string
 */
const getDefaultTabLabel = (searchParams) => {
  return tabLabels.includes(searchParams.view)
    ? searchParams.view
    : tabLabels[0];
};
```